.ci/release/Jenkinsfile

```diff
@@ -0,0 +1,234 @@
+          steps {
+            // If this fails, an exception should be thrown and execution will halt
+            script {
+              def r = sh(label: "Check Maven status", script: "scripts/jenkins/check_maven.sh -u https://status.maven.org/api/v2/summary.json --component OSSRH", returnStatus: true)
+              if (r == 1) {
+                error("Failing release build because Maven is the OSSRH component is not fully operational. See https://status.maven.org/ for more details.")
+              }
+            }
+          }
+        }
+        stage('Check master build status') {
+          steps {
+            script {
+              // If this build is not green: https://apm-ci.elastic.co/job/apm-agent-java/job/apm-agent-java-mbp/job/master/
+              if(!buildStatus(host: 'apm-ci.elastic.co', job: ['apm-agent-java', 'apm-agent-java-mbp', 'master'], return_boolean: true)) {
+                input(message: "WARNING! The master build is not passing. Do you wish to continue?")
+              }
+            }
+          }
+        }
+        stage('Require confirmation that CHANGELOG.asciidoc has been updated') {
+          steps {
+            input(message: """
+            Update CHANGELOG.asciidoc to reflect the new version release:
+            Go over PRs or git log and add bug fixes and features.
+            Move release notes from the Unreleased sub-heading to the correct [[release-notes-{major}.x]] sub-heading (Example PR for 1.13.0 release).
+
+            Click 'Proceed' to confirm that this step has been completed or Abort to stop the build in order to complete this step.
+            """
+            )
+          }
+        }
+        stage('Review project version') {
+          steps {
+            script {
+              def ver = mvnVersion(showQualifiers: true)
+              def should_continue = input(message: "Current version is ${ver}", parameters: [
+                [
+                  $class: 'ChoiceParameterDefinition',
+                  name: "Do you wish to update the version?",
+                  "choices": ["Yes", "No"],
+                  description: "Selecting 'Yes' will allow you to select the new version in the next step."
+                ]
+              ])
+              if (should_continue == 'Yes'){
+                def new_version = input(message: "Please enter version to change to:", parameters:
+                  [
+                    [
+                      $class: 'StringParameterDefinition',
+                      defaultValue: "${ver}",
+                      description: 'We will run mvn versions:set -DnewVersion=<NEW_VERSION>', name: 'New Version'
+                    ]
+                  ]
+                )
+                sh(name: "mavenVersionUpdate", script: "mvn --batch-mode release:update-versions -DdevelopmentVersion=${new_version}")
+                withGitRelease() {
+                  sh(script: "git commit -a -m 'Version bump'")
```

Comment:
don't forget to push :)

Reply:
The reason I didn't push here is that it happens in the step shortly after. Do you think we should have a push here as well? I'm fine with either. :)

Reply:
If we don't push here, the internal CI doesn't see the changes

Reply:
Ah! You're right of course. I'll make this change.